pysr/sr.py

```diff
@@ -2586,3 +2583,122 @@ def _mutate_parameter(param_name: str, param_value):
         return False
 
     return param_value
+
+
+class PySRSequenceRegressor(PySRRegressor):
+    def __init__(
+        self,
+        recursive_history_length: int = 0,
+        **kwargs,
+    ):
+        self.recursive_history_length = recursive_history_length
+        super().__init__(**kwargs)
+
+    def fit(
+        self,
+        X,
+        weights=None,
+        variable_names: Optional[ArrayLike[str]] = None,
+        complexity_of_variables: Optional[
+            Union[int, float, List[Union[int, float]]]
+        ] = None,
+        X_units: Optional[ArrayLike[str]] = None,
+    ) -> "PySRRegressor":
+        """
+        Search for equations to fit the time series dataset and store them in `self.equations_`.
+
+        Parameters
+        ----------
+        X : ndarray | pandas.DataFrame
+            Training time series data of shape (n_times, ...).
```

Comment:
Since PySRRegressor currently only handles 2D data of shape `(n_rows, n_features)`, this should be a fixed shape `(n_times, n_features)`

Reply:
I added functionality to support any dimensional data :)

Reply:
Currently the preprocessor is designed to give variables names like `x_20_16t_3`
This variable name means if you looked at the t-3 th data point, and looked at the index [20][16], you would find the variable
With 2d arrays we still need to concatenate and flatten

Reply:
Cool!

I guess I do think it should just force the user to flatten their data along the second axis though. Automatically flattening input data is a bit of a footgun in case the user provides images with different dimension than trained on. Better to force the user to be very explicit about the number of features they have. It's a slight inconvenience but it's much safer.

Basically, just have an assertion like

```python
assert len(X.shape) == 2
```

would be good enough. I think scikit-learn actually has some utility that does this automatically.